apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1610,6 +1627,10 @@ func (u *UniterAPI) LeaveScope(args params.RelationUnits) (params.ErrorResults,
 
 // ReadSettings returns the local settings of each given set of
 // relation/unit.
+//
+// NOTE(achilleasa): Using this call to read application data is deprecated
+// and will not work for k8s charms (see LP1876097). Instead, clients should
+// use ReadLocalApplicationSettings.
```

Comment:
can we make this not available in version 16 then?

Reply:
No you still need it for things to work when you upgrade your controller and before you upgrade your models (the not-yet-upgraded agents still make use of the call).

Reply:
Non-upgraded models will be using version 15, so it'd still be available there.  Just not version 16 and beyond.  We have ways to say, this method is only in version 16 and beyond, why not the opposite?

Reply:
Note that the method is deprecated for reading the application data-bag _only_. It is still used for reading the unit's data bag (which was its original purpose AFAICT). Therefore, we cannot remove it.

Reply:
I wasn't clear on the overloading here.